USF-1975: Synchronize Checkout drop-in documentation (2) (#200)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/checkout/containers/overview.mdx

---
title: Checkout containers overview
description: Learn how to use the containers for the checkout drop-in component.
---

import OptionsTable from '@components/OptionsTable.astro';

> During the checkout process there are containers that need to be rendered or not depending on the use of some conditions, in our case, data extracted from the cart state.

Reviewer: ```suggestion
During the checkout process, some containers need to be rendered or not depending on the use of some conditions, in our case, data extracted from the cart state.
```

## ConditionalContainer configurations

The `ConditionalContainer` container provides the following configuration options:

<OptionsTable
  compact
  options={[
    ['Option', 'Type', 'Req?', 'Description'],
    ['hideOnEmptyCart', 'boolean', 'No', 'Hides the container if the cart is empty (default value is true).'],
    ['hideOnVirtualCart', 'boolean', 'No', 'Hides the container if the cart is virtual (default value is false).'],
  ]}
/>

The structure of the `withConditionalRendering()` function is defined as following:

```ts
export interface ConditionalProps {
  hideOnEmptyCart?: boolean;
  hideOnVirtualCart?: boolean;
}

export const withConditionalRendering = <P extends object>(
  WrappedContainer: Container<P>
) => {
  const ConditionalContainer = ({
    hideOnEmptyCart = true,
    hideOnVirtualCart = false,
    ...props
  }: ConditionalProps & P) => {
    // Logic to hide the container goes here
  };

  return ConditionalContainer;
};
```

:::note[Checkout containers]

> As mentioned, Checkout drop-in is composed of some containers that need to be rendered based on cart coditions. In the following pages, we will describe all the Checkout containers indicating whether they need conditional rendering or not, therefore whether they accept both `hideOnEmptyCart` and `hideOnVirtualCart` props.

Reviewer: ```suggestion
As mentioned, the Checkout drop-in consists of several containers rendered based on cart conditions. In the following pages, we will describe all the Checkout containers indicating whether they need conditional or whether they accept both `hideOnEmptyCart` and `hideOnVirtualCart` props.
```
